origins_workflow/src/Controller/AuditController.php

```diff
@@ -0,0 +1,139 @@
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Controller\ControllerBase;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Controller to handle auditing of nodes.
+ */
+class AuditController extends ControllerBase implements ContainerInjectionInterface {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * A logger instance.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $account;
+
+  /**
+   * Creates a new ModerationStateConstraintValidator instance.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   The logger interface.
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   The current user.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, LoggerInterface $logger, AccountInterface $account) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->logger = $logger;
+    $this->account = $account;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('logger.factory')->get('origins_workflow'),
+      $container->get('current_user')
+    );
+  }
+
+  /**
+   * Content Audit.
+   *
+   * @return string
+   *   Return confirmation string.
+   */
+  public function contentAudit($nid) {
+    $render_array = [];
+    $msg = $this->t('Invalid node ID - @nid', ['@nid' => $nid]);
+    if (!empty($nid)) {
+      $node = $this->entityTypeManager()->getStorage('node')->load($nid);
+      if ($node) {
+        // Retrieve audit text from config.
+        $audit_button_text = $this->config('origins_workflow.auditsettings')->get('audit_button_text');
+        $audit_confirmation_text = $this->config('origins_workflow.auditsettings')->get('audit_confirmation_text');
+        // Show confirmation text to user.
+        $render_array['confirmation_text'] = [
+          '#markup' => $this->t($audit_confirmation_text),
+          '#prefix' => "<div class='confirmation_text'>",
+          '#suffix' => "</div>",
+          '#weight' => 0,
+        ];
+        // Build a confirm link.
+        $link_object = Link::createFromRoute($this->t($audit_button_text),
+          'origins_workflow.audit_controller_confirm_audit',
+          ['nid' => $nid],
+          ['attributes' => ['rel' => 'nofollow', 'class' => 'audit_link']]);
+        // Add confirm link to render array.
+        $render_array['link1'] = array_merge($link_object->toRenderable(),
```

Comment:
Why not use the link render element?

Reply:
Not sure what you mean, please could you explain ?

Reply:
Sure, if you use https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Render%21Element%21Link.php/class/Link/8.7.x you don't need to create 2 separate elements and merge the arrays.